src/neural-network.js

```diff
@@ -292,17 +292,10 @@ export default class NeuralNetwork {
    *       momentum: (number),
    *       activation: ['sigmoid', 'relu', 'leaky-relu', 'tanh']
    */
-  updateTrainingOptions(opts) {
-    if (opts.iterations) { this.trainOpts.iterations = opts.iterations; }
-    if (opts.errorThresh) { this.trainOpts.errorThresh = opts.errorThresh; }
-    if (opts.log) { this._setLogMethod(opts.log); }
-    if (opts.logPeriod) { this.trainOpts.logPeriod = opts.logPeriod; }
-    if (opts.learningRate) { this.trainOpts.learningRate = opts.learningRate; }
-    if (opts.momentum) { this.trainOpts.momentum = opts.momentum; }
-    if (opts.callback) { this.trainOpts.callback = opts.callback; }
-    if (opts.callbackPeriod) { this.trainOpts.callbackPeriod = opts.callbackPeriod; }
-    if (opts.timeout) { this.trainOpts.timeout = opts.timeout; }
-    if (opts.activation) { this.activation = opts.activation; }
+  _updateTrainingOptions(opts) {
+    Object.keys(NeuralNetwork.trainDefaults).forEach(opt => this.trainOpts[opt] = opts[opt] || this.trainOpts[opt]);
```

Comment:
I think this would create weird behavior in the instance of updating training options.  For example:

```
net.train(data, { log: true });
net.train(data2, { logPeriod: 50 });
```
The second training would not print because the log would be reset to the default.  The same problem would occur any time the update training options method is called.

I like the idea of iterating instead of static if's but maybe we need to iterate on the union of keys from defaults and the passed in objects.

Reply:
I'm iterating over the keys of the static, not the static itself. I'm still setting the `this.trainOpts`

Reply:
ideally this would throw an error, as the net is still training

Reply:
@robertleeplummerjr , I think this is assuming synchronous training, so there shouldn't be any overlap with train timing.

Reply:
doh, that is correct.

Reply:
all the new tools easily confuse me

Reply:
SUPPER SORRY.  I read `... = opts[opt] || this.trainOpts[opt]` and thought `... = opts[opt] || this.trainDefaults[opt]`.  This is fine the way it is given that we call this in the constructor:

```
this._updateTrainingOptions(Object.assign({}, this.constructor.trainDefaults, options));
```

![tenor](https://user-images.githubusercontent.com/2294139/35595066-a461ee1c-05d2-11e8-8b3f-1157cbb0db0f.gif)